content/blog/insecure-software-development/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Insecure Software Development
+date: "2022-01-26T21:00:00.284Z"
+description: "A tale of software development and hidden complexity"
+---
+
+## TL;DR;
+
+If you don't have the time or is not sure if you want to read this post, here are the main points discussed:
+
+1. Hidden complexity is a common cause of security vulnerabilities
+2. There are multiple ways to tackle complexity in codebases, each with different degrees of success and pitfalls
+3. Code review is a great tool to reduce complexity before it becomes a problem
+4. Type juggling is still a thing
+
+## Introduction
+
+Do you remember `Log4J`? What about `Dependency Confusion`? How could you not? At the time of this writing both happened over this past year and disrupted parts of our industry. Development and Infrastructure teams scrambled to patch their systems. Companies with poor asset management had it even worse, trying to identify and uncover vulnerable infrastructure while responding to these incidents. So what does both of these incidents
+have in common? I will blame _hidden complexity_ here.
+
+Log4J abstracted environment variable parsing in log messages through a custom syntax that is parsed at the time of logging. Dependency confusion was caused by package managers trying to abstract the domain of dependency management, which is far from simple. In both cases library maintainers made the behaviour implicit.
+
+## Complexity
```

Comment:
I think throughout this blog post, you have discussed how complexity happens, how to reduce it, and how to have the right amount of it, but have never defined what complexity really is. Complexity could have multiple definitions, both technical and non-technical, so I think for the sake of clarity, as well as not confusing a more beginner audience (if you do care about writing for that audience), I think you should *define* what complexity is as well.

Reply:
That's a good point, I will think about it and adjust the post. :)